diff --git a/src/FSAL/commonlib.c b/src/FSAL/commonlib.c
index 66de46d0a..009b91da8 100644
--- a/src/FSAL/commonlib.c
+++ b/src/FSAL/commonlib.c
@@ -1266,6 +1266,10 @@ int populate_posix_file_systems(bool force)
 	while ((mnt = getmntent(fp)) != NULL) {
 		if (mnt->mnt_dir == NULL)
 			continue;
+		struct stat st;
+		if (stat(mnt->mnt_dir, &st) == 0 && !S_ISDIR(st.st_mode)) {
+			continue;
+		}
 
 		posix_create_file_system(mnt);
 	}
