diff --git a/src/FSAL/commonlib.c b/src/FSAL/commonlib.c
index 2df85a1..2738ba7 100644
--- a/src/FSAL/commonlib.c
+++ b/src/FSAL/commonlib.c
@@ -64,6 +64,7 @@
 #ifdef HAVE_MNTENT_H
 #include <mntent.h>
 #endif
+#include "gsh_config.h"
 #include "gsh_list.h"
 #ifdef USE_BLKID
 #include <blkid/blkid.h>
@@ -970,6 +971,13 @@ static bool posix_get_fsid(struct fsal_filesystem *fs)
 
 	fs->dev = posix2fsal_devt(mnt_stat.st_dev);
 
+	if (nfs_param.core_param.fsid_device) {
+		fs->fsid_type = FSID_DEVICE;
+		fs->fsid.major = fs->dev.major;
+		fs->fsid.minor = fs->dev.minor;
+		return true;
+	}
+
 #ifdef USE_BLKID
 	dev_name = blkid_devno_to_devname(mnt_stat.st_dev);
 
diff --git a/src/config_samples/config.txt b/src/config_samples/config.txt
index 687fb16..21e3bf8 100644
--- a/src/config_samples/config.txt
+++ b/src/config_samples/config.txt
@@ -142,6 +142,8 @@ NFS_CORE_PARAM {}
 
 	heartbeat_freq(uint32, range 0 to 5000 default 1000)
 
+	fsid_device(bool, default false)
+
 NFS_IP_NAME {}
 --------------
 
diff --git a/src/include/gsh_config.h b/src/include/gsh_config.h
index 9fbc3a4..0e7919c 100644
--- a/src/include/gsh_config.h
+++ b/src/include/gsh_config.h
@@ -390,6 +390,8 @@ typedef struct nfs_core_param {
 	char *ganesha_modules_loc;
 	/* Frequency of dbus health heartbeat in ms. Set to 0 to disable */
 	uint32_t heartbeat_freq;
+	/* Whether to use device major/minor for fsid. Defaults to false. */
+	bool fsid_device;
 } nfs_core_parameter_t;
 
 /** @} */
diff --git a/src/support/nfs_read_conf.c b/src/support/nfs_read_conf.c
index a40324f..228c6a2 100755
--- a/src/support/nfs_read_conf.c
+++ b/src/support/nfs_read_conf.c
@@ -184,6 +184,8 @@ static struct config_item core_params[] = {
 		       nfs_core_param, ganesha_modules_loc),
 	CONF_ITEM_UI32("heartbeat_freq", 0, 5000, 1000,
 		       nfs_core_param, heartbeat_freq),
+	CONF_ITEM_BOOL("fsid_device", false,
+		       nfs_core_param, fsid_device),
 	CONFIG_EOL
 };
 
